### DISCUSSION TITLE
My first deployment.

### DISCUSSION THREAD

**Discussion by AliAlnajjar** [saleor/saleor#15273] [Deployment]
I am completely new to Saleor and to cloud. I was trying to deploy a docker image from, github image registry to digitalocean through the "app" thing in digitalocean. Anyone here can explain me how to do that to get a saleor instance running

**Comment by jorgdec** [comment]
This discussion is pretty old, but this might help other people struggling with setting it up.

This is what I did to deploy a (minimal) Saleor instance through the digital ocean app platform.

I have 2 services
- Core based on [saleor/saleor image](https://github.com/saleor/saleor/pkgs/container/saleor)
- Dashboard based on [saleor/saleor-dashboard image](https://github.com/saleor/saleor/pkgs/container/saleor)

and 1 postgresql database

## Setup
The following things you should configure to make sure the deployment and health check succeeds.

### ENV variables
The core service should have (at least) the following env variables configured
- DATABASE_URL -> this variable will normally get generated by app platform. Example would be  ${postgresql-db.DATABASE_URL}
- ALLOWED_HOSTS -> this defaults to localhost and make the application not work. You can use * to get up and running and finetune this later
- SECRET_KEY -> Used by django to generate hashes, should be a random string
- DASHBOARD_URL -> this should be the URL to access your dashboard service. Example would be https://saleor-abcde.ondigitalocean.app/dashboard

The dashboard service should have (at least) the following env variables configured
- API_URL -> this should be the url to your graphql endpoint. Example would be https://saleor-abcde.ondigitalocean.app/graphql/

### HTTP Request Routes
You need to add 1 route to each service
Core
- /

Dashboard
- /dashboard

### Ports
Core -> 8000
Dashboard -> 80

### Healthchecks
Core -> http on /
Dashboard -> http on /dashboard

### Migrations
I had some issues with running the migrations when deploying, but i fixed it by adding the following startup command to the core service. (i'm not sure if this is the best solution).

`bash -c "python manage.py migrate --noinput && uvicorn saleor.asgi:application --host=0.0.0.0 --port=8000 --workers=2 --lifespan=off --ws=none --no-server-header --no-access-log --timeout-keep-alive=35 --timeout-graceful-shutdown=30 --limit-max-requests=10000" `

### Version
Make sure to verify that the core and dashboard are on the same minor version. Otherwise, you'll [see a lot of errors after loggin in](https://github.com/saleor/saleor-dashboard/issues/5944)
